packages/odyssey-react-mui/src/RadioGroup.tsx

```diff
@@ -0,0 +1,102 @@
+/*!
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import {
+  FormControl,
+  FormLabel,
+  FormHelperText,
+  visuallyHidden,
+  Radio,
+} from "./";
+import { RadioGroup as MuiRadioGroup } from "@mui/material";
+import { useMemo } from "react";
+
+export interface RadioGroupProps {
+  /**
+   * The name of the radio group, which only needs to be changed
+   * if there are multiple radio groups on the same screen
+   */
+  name?: string;
```

Comment:
Should we make this required for the sake of hygiene?

I don't want to force devs to do extra work, but it might make automated testing and maintenance easier if these are unique from go.

Reply:
It's not required in HTML, and instead of `name`, our tests should be targeting ARIA attributes like `label` and `role` anyway.

Reply:
`name` isn't required for other inputs, but it's needed for `type="radio"` functionality to behave correctly, and it needs to be unique across radio groups.

For ex, if a dev leaves this unset in two fields, validation and selection will be shared across their inputs, even if they are in different components.

Reply:
I'm temporarily leaving it as optional, defaulting to "radio," for now. I want to avoid the complexity of generating a unique ID (unless that's simple to do?) and assume it's on the consumer to know if they need to specify a `name` or not.

I assume, if a consumer team is bringing their own preexisting form, they'll already have some unique ID defined in their JSON schema if they have multiple radio groups. If they don't, it should be because they don't need it, in which case the default "radio" should be okay. Is that right?

Reply:
Yup, as long as there's a single `radio`, they won't run into issues!

Reply:
We have a `useUniqueId` hook which will generate a unique one for you. It's pretty simple to use:
```js
const uniqueId = useUniqueId()
const uniqueName = useMemo(() => (name ? name : uniqueId), [name, uniqueId])
```

But we don't have to do that either. Can we just leave it blank like HTML? It's fine if they forget it. The intent is that they'll pass one, but it's optional.